prisma-fmt/src/code_actions/relations.rs

```diff
@@ -242,3 +177,45 @@ pub(super) fn add_referenced_side_unique(
 
     actions.push(CodeActionOrCommand::CodeAction(action));
 }
+
+fn create_missing_unique<'a>(
+    schema: &str,
+    model: ModelWalker<'a>,
+    mut fields: impl ExactSizeIterator<Item = ScalarFieldWalker<'a>> + 'a,
+) -> TextEdit {
+    let (new_text, range) = if fields.len() == 1 {
+        let new_text = String::from(" @unique");
+
+        let field = fields.next().unwrap();
+        let position = crate::position_after_span(field.ast_field().span, schema);
+
+        let range = Range {
+            start: position,
+            end: position,
+        };
+
+        (new_text, range)
+    } else {
+        let fields = fields.map(|f| f.name()).collect::<Vec<_>>().join(", ");
+
+        let indentation = model.indentation();
+        let newline = model.newline();
+
+        let separator = if model.ast_model().attributes.is_empty() {
+            newline.as_ref()
+        } else {
+            ""
+        };
+
+        let new_text = format!("{separator}{indentation}@@unique([{fields}]){newline}}}");
```

Comment:
Getting the formatting _just right_™ is a lot of implementation effort and extra complexity, I'm really wondering whether we shouldn't take a more brute-force approach: append the thing at the end of the block, and as an edit, replace the whole model with the model + the addition passed through the reformatter.

Reply:
I've been thinking the same... a lot. What I'm worried about is nobody else does that. It's a bit weird to get a reformatted model only when adding a unique constraint. It would make things easier still...

Reply:
rust-analyzer doesn't try hard to match your formatting, from the looks of it

Reply:
Especially when reformatting adds relation fields and all that...

Reply:
it would be worth looking into what other language server implementations do